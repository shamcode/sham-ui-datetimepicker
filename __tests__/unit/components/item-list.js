import ItemList  from '../../../src/components/item-list.sfc';
import renderer from 'sham-ui-test-helpers';
import { onclick } from 'sham-ui-directives';

it( 'renders correctly', () => {
    const meta = renderer( ItemList, {
        today: new Date( 2020, 3, 25, 15, 3, 30 ),
        value: new Date( 2020, 3, 23, 13, 2, 17 ),
        date: new Date( 2020, 3, 23, 13, 2, 17 )
    }, {
        directives: {
            onclick
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
