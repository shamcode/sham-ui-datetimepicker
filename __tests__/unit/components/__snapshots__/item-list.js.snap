// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "change": [Function],
    "classForDate": [Function],
    "clickOnHeader": [Function],
    "date": 2020-04-23T06:02:17.000Z,
    "header": [Function],
    "isDateSelectable": [Function],
    "items": [Function],
    "nextDelta": 1,
    "nextSymbol": "▸",
    "prevDelta": -1,
    "prevSymbol": "◂",
    "select": [Function],
    "showChange": true,
    "today": 2020-04-25T08:03:30.000Z,
    "value": 2020-04-23T06:02:17.000Z,
  },
  "html": "
<thead>
  <tr>
    <th><span>◂</span></th>
    <th colspan=\\"5\\" class=\\"picker-switch\\">null</th>
    <th><span>▸</span></th>
  </tr>
</thead>
<tbody>
  <tr>
    <td colspan=\\"7\\"></td>
  </tr>
</tbody>
",
}
`;
